tools/shared.py

```diff
@@ -548,7 +548,7 @@ def check_sanity(force=False):
       logging.critical('The JavaScript shell used for compiling (%s) does not seem to work, check the paths in %s' % (COMPILER_ENGINE, EM_CONFIG))
       sys.exit(1)
 
-    if NODE_JS != COMPILER_ENGINE:
+    if NODE_JS != COMPILER_ENGINE and Settings.RUNNING_JS_OPTS:
```

Comment:
Yes, that var is set when compiling a particular file.

What we can do is not warn about node.js at all during setup. And during compile time, we'd need a good error message if it's actually needed. I believe that is exactly what we do for java (for closure compiler), so basically this change could handle node the way java is.

You will probably need to modify tests in `test/test_sanity.py` for such a change.